python/ray/util/spark/start_ray_node.py

```diff
@@ -54,63 +54,42 @@
     process = subprocess.Popen([ray_cli_cmd, "start", *arg_list], text=True)
 
     def try_clean_temp_dir_at_exit():
-        try:
-            # Wait for a while to ensure the children processes of the ray node all
-            # exited.
-            time.sleep(SIGTERM_GRACE_PERIOD_SECONDS + 0.5)
-            if process.poll() is None:
-                # "ray start ..." command process is still alive. Force to kill it.
-                process.kill()
-
-            # Release the shared lock, representing current ray node does not use the
-            # temp dir.
-            fcntl.flock(lock_fd, fcntl.LOCK_UN)
-
-            try:
-                # acquiring exclusive lock to ensure copy logs and removing dir safely.
-                fcntl.flock(lock_fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
-                lock_acquired = True
-            except BlockingIOError:
-                # The file has active shared lock or exclusive lock, representing there
-                # are other ray nodes running, or other node running cleanup temp-dir
-                # routine. skip cleaning temp-dir, and skip copy logs to destination
-                # directory as well.
-                lock_acquired = False
-
-            if lock_acquired:
-                # This is the final terminated ray node on current spark worker,
-                # start copy logs (including all local ray nodes logs) to destination.
-                if collect_log_to_path:
-                    try:
-                        copy_log_dest_path = os.path.join(
-                            collect_log_to_path,
-                            os.path.basename(temp_dir) + "-logs",
-                            socket.gethostname(),
-                        )
-                        ray_session_dir = os.readlink(
-                            os.path.join(temp_dir, "session_latest")
-                        )
-                        shutil.copytree(
-                            os.path.join(ray_session_dir, "logs"),
-                            copy_log_dest_path,
-                        )
-                    except Exception as e:
-                        _logger.warning(
-                            "Collect logs to destination directory failed, "
-                            f"error: {repr(e)}."
-                        )
-
-                # Start cleaning the temp-dir,
-                shutil.rmtree(temp_dir, ignore_errors=True)
-        except Exception:
-            # swallow any exception.
-            pass
-        finally:
-            fcntl.flock(lock_fd, fcntl.LOCK_UN)
-            os.close(lock_fd)
+        _try_clean_temp_dir_at_exit(
+            process=process,
+            collect_log_to_path=collect_log_to_path,
+            temp_dir=temp_dir,
```

Comment:
For global mode, can we get correct temp_dir value `/local_disk0/tmp/ray` here ? I don't see code that sets temp_dir variable in global mode case.

Reply:
It's set here https://github.com/ray-project/ray/blob/648c8ad86b1a22133cc47fbe5782b538e55aed35/python/ray/util/spark/cluster_init.py#L553 then pass to the command here https://github.com/ray-project/ray/blob/648c8ad86b1a22133cc47fbe5782b538e55aed35/python/ray/util/spark/cluster_init.py#L570